tensorbay/opendataset/nuImages/loader.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import base64
+import os
+from typing import Any, Dict, List, Optional
+
+from ...dataset import Data, Frame, FusionDataset, FusionSegment
+from ...label import LabeledBox2D, LabeledRLE
+from ...sensor import Camera
+from .._utility.nuScenes import (
+    get_info_with_determined_token,
+    get_info_with_token,
+    transpose_rle,
+    uncompress_rle,
+)
+
+DATASET_NAME = "nuImages"
+
+
+def nuImages(path: str) -> FusionDataset:
+    """Dataloader of the `nuScenes`_ dataset.
+
+    .. _nuImages: https://www.nuscenes.org/nuimages
+
+    The file structure should be like::
+
+        <path>
+            nuimages-v1.0-all-metadata/
+                v1.0-mini/
+                    attribute.json
+                    calibrated_sensor.json
+                    category.json
+                    ego_pose.json
+                    instance.json
+                    log.json
+                    object_ann.json
+                    sample_data.json
+                    sample.json
+                    sensor.json
+                    surface_ann.json
+                v1.0-test/
+                    ...
+                v1.0-train/
+                    ...
+                v1.0-val/
+                    ...
+            samples/
+                CAM_BACK/
+                CAM_BACK_LEFT/
+                CAM_BACK_RIGHT/
+                CAM_FRONT/
+                CAM_FRONT_LEFT/
+                CAM_FRONT_RIGHT/
+            sweeps/
+                CAM_BACK/
+                CAM_BACK_LEFT/
+                CAM_BACK_RIGHT/
+                CAM_FRONT/
+                CAM_FRONT_LEFT/
+                CAM_FRONT_RIGHT/
+            nuimages-v1.0-mini/
+                samples/
+                    ...
+                sweeps/
+                    ...
+                v1.0-mini/
+                   ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = FusionDataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    metadata_path = os.path.join(root_path, "nuimages-v1.0-all-metadata")
+    for subset in os.listdir(metadata_path):
+        if subset != "v1.0-mini":
+            continue
+        dataset.add_segment(_get_segment(root_path, subset, metadata_path))
+    return dataset
+
+
+def _get_segment(root_path: str, subset: str, metadata_path: str) -> FusionSegment:
+    is_test = subset.endswith("test")
+    annotation_info = _get_annotation_info(os.path.join(metadata_path, subset), is_test)
+    segment = FusionSegment(subset)
+    _load_frame_and_sensor(segment, annotation_info, root_path, is_test)
```

Comment:
is that possible to return segment from `_load_frame_and_sensor` rather than input a segment instance and modify it?